core/deployment/src/main/java/io/quarkus/deployment/index/ApplicationArchiveBuildStep.java

```diff
@@ -171,47 +169,41 @@ ApplicationArchivesBuildItem build(
         return ret;
     }
 
-    private static Collection<? extends Path> getMarkerFilePaths(ClassLoader classLoader, Set<String> applicationArchiveFiles,
-            ArchiveRootBuildItem root)
+    private static void addMarkerFilePaths(Set<String> applicationArchiveFiles,
+            ArchiveRootBuildItem root, CurateOutcomeBuildItem curateOutcomeBuildItem, Set<Path> depsToIndex)
             throws IOException {
-        List<Path> ret = new ArrayList<>();
-        for (String file : applicationArchiveFiles) {
-            Enumeration<URL> e = classLoader.getResources(file);
-            while (e.hasMoreElements()) {
-                final URL url = e.nextElement();
-                final Path path = urlToPath(url, file);
-                if (!root.isExcludedFromIndexing(path)) {
-                    ret.add(path);
+        for (AppDependency dep : curateOutcomeBuildItem.getEffectiveModel().getFullDeploymentDeps()) {
```

Comment:
This may need adjustment. Previously it used to look for the markers in the deployment classloader, which includes more resources than the list of the deployment deps the resolver returns.

Reply:
Yea, this needs to be user deps, not deployment deps.

Reply:
Done.